localflavor/gr/forms.py

```diff
@@ -60,3 +61,56 @@ def clean(self, value):
         if mod != check:
             raise ValidationError(self.error_messages['invalid'])
         return val
+
+
+class GRSocialSecurityNumberCodeField(Field):
+    """
+    Greek social security number (AMKA) field.
+
+    The allow_test_value option can be used to enable the usage of the
+    non valid 00000000000 (11 zeros) value for testing and development
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid greek social security number (AMKA - 11 digits).'),
+    }
+
+    def __init__(self, allow_test_value=False, *args, **kwargs):
+        self.allow_test_value = allow_test_value
+        super(GRSocialSecurityNumberCodeField, self).__init__(*args, **kwargs)
+
+    def check_date(self, val):
+        try:
+            datetime.datetime.strptime(val[:6], '%d%m%y')
+        except:
+            raise ValidationError(self.error_messages['invalid'])
+
+    def check_lunh(self, val):
```

Comment:
Please use the `lunh` function that's already in localflavor instead of including your own version.

https://github.com/django/django-localflavor/blob/d23320486862d993070de51ef92e31b8f4cc61a3/localflavor/generic/checksums.py#L10

Reply:
Ok